os/net/netdev/netdev_dhcpd.c

```diff
@@ -0,0 +1,165 @@
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>		/* TinyAra configuration */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <unistd.h>
+#include <time.h>
+#include <errno.h>
+#include <debug.h>				/* For ndbg, vdbg */
+
+#include <net/if.h>
+#include <netinet/in.h>
+#include <semaphore.h>
+#include <pthread.h>
+#include <netutils/netlib.h>
+#include <protocols/dhcpd.h>	/* Advertised DHCPD APIs */
+
+#include <net/lwip/netif.h>
+#include <net/lwip/netifapi.h>
+
+/****************************************************************************
+ * Global Data
+ ****************************************************************************/
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+#define DHCP_SERVER_PORT         67
+#define DHCP_CLIENT_PORT         68
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Global Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: dhcps_server_status
+ ****************************************************************************/
+int netdev_dhcp_server_status(char *intf)
+{
+	struct netif *cur_netif;
+	cur_netif = netif_find(intf);
+	if (cur_netif == NULL) {
+		ndbg("[DHCPS] no network interface for dhcpd start\n");
+		return 0;
+	}
+
+	if (cur_netif->dhcps_pcb == NULL) {
+		ndbg("[DHCPS] DHCP server closed\n");
+		return 0;
+	} else {
+		ndbg("[DHCPS] DHCP server opened\n");
+		return 1;
+	}
+}
+
+/****************************************************************************
+ * Name: dhcps_server_start
+ ****************************************************************************/
+int netdev_dhcp_server_start(char *intf, dhcp_sta_joined dhcp_join_cb)
+{
+	struct netif *cur_netif;
+	cur_netif = netif_find(intf);
+	if (cur_netif == NULL) {
+		ndbg("[DHCPS] no network interface for dhcpd start\n");
+		return -1;
+	}
+	if (dhcp_join_cb) {
+		if (dhcps_register_cb(dhcp_join_cb) != ERR_OK) {
+			ndbg("[DHCPS] link callback fail\n");
+			return -1;
+		}
+	}
+
+	if (netifapi_dhcps_start(cur_netif) == ERR_OK) {
+		ndbg("[DHCPS] started successfully (LWIP)\n");
+		return OK;
```

Comment:
You using OK here, then you can use ERROR instead of '-1' because it means you already include type.h

Reply:
Applied. Thanks for the comment.